packages/frontend/src/components/accounts/two_factor/TwoFactorVerifyModal.js

```diff
@@ -1,32 +1,125 @@
+import BN from 'bn.js';
+import { utils } from 'near-api-js';
 import React, { useEffect, useState } from 'react';
 import { Translate } from 'react-localize-redux';
 import { useDispatch, useSelector } from 'react-redux';
 import styled from 'styled-components';
 
 import { Mixpanel } from '../../../mixpanel/index';
-import { resendTwoFactor, get2faMethod } from '../../../redux/actions/account';
-import { selectAccountSlice } from '../../../redux/slices/account';
+import { resendTwoFactor, get2faMethod, getMultisigRequest } from '../../../redux/actions/account';
+import { selectAccountSlice, selectAccountMultisigRequest } from '../../../redux/slices/account';
 import { selectActionsPending, selectStatusSlice } from '../../../redux/slices/status';
 import { WalletError } from '../../../utils/walletError';
+import AlertBanner from '../../common/AlertBanner';
 import FormButton from '../../common/FormButton';
 import Modal from '../../common/modal/Modal';
 import ModalTheme from '../ledger/ModalTheme';
 import TwoFactorVerifyInput from './TwoFactorVerifyInput';
 
+const {
+    format: {
+        formatNearAmount
+    },
+    key_pair: {
+        PublicKey,
+        KeyType
+    },
+} = utils;
+
 const Form = styled.form`
     display: flex;
     flex-direction: column;
     align-items: center;
     width: 100%;
 `;
 
+const formatNear = (amount) => {
+    try {
+        return formatNearAmount(amount, 4);
+    } catch {
+        // Near amount may be stored as hex rather than decimal
+        return formatNearAmount(new BN(amount, 16).toString(), 4);
+    }
+};
+
+const getTranslationsFromMultisigRequest = ({ actions, receiverId, accountId }) => {
```

Comment:
This is a great function and feels like it might be used in other places outside of the two factor modal. Should we put it in a helper, etc?

Reply:
Yep definitely makes sense as a helper! Do you think we should keep the function as is, i.e. still returning the `<Translation />` data, or should it perhaps be more generic by returning the `multisigRequest` in a more usable format? If it was more generic we'd most likely need to map it again here which could still bloat this component file 🤔

Reply:
It's also worth noting that `near-contract-helper` also covers messaging around setting up 2FA for the first time (`requestId === -1`) but since we've disabled that flow for SMS i've left that out.

Also, initial verification is done within `<EnterVerificationCode />` rather than this modal so it wouldn't be needed in this component, but may be needed within this new helper if we chose to eventually add something similar to `<EnterVerificationCode />` 🤔

Reply:
I like it the way it is because it'd be easy to just plug it into any UI and yes additional bloat seems not worth it IMO 👍